### PR TITLE
Use parse-author only if the field is a string

### LINKED ISSUE #252: Cannot read property \'url\' of undefined\n    at readMeta.then.pkg
**What version of `electron-installer-windows` are you using?**
Most recent

**What version of `node` and `npm` are you using?**
6.5.0

**What operating system are you using?**
Windows 10

**Can you compile the [example app](https://github.com/electron-userland/electron-installer-windows/tree/master/example) successfully?**
Yes

**What did you do? Please include the configuration you are using for `electron-installer-windows`.**
`$ electron-installer-windows --src radiam-agent-win32-x64/ --dest ~/Desktop/`

**What did you expect to happen?**
Package to get built

**What actually happened?**
```
Creating package (this may take a while)
TypeError: Cannot read property 'url' of undefined
    at readMeta.then.pkg (C:\Users\axfel\AppData\Roaming\npm\node_modules\electron-installer-windows\src\installer.js:71:22)
    at <anonymous> 'TypeError: Cannot read property \'url\' of undefined\n    at readMeta.then.pkg (C:\\Users\\axfel\\AppData\\Roaming\\npm\\node_modules\\electron-installer-windows\\src\\installer.js:71:22)\n    at <anonymous>'
```

### PR DESCRIPTION
**Pull request by fcastilloec**
This fixes #252

### CODE REVIEW

File: src/installer.js [hunk 1]
```diff
@@ -118,7 +118,8 @@ class SquirrelInstaller extends common.ElectronInstaller {
     return common.readMetadata(this.userSupplied)
       .then(pkg => {
         pkg = pkg || {}
-        const authors = [parseAuthor(pkg.author).name]
+        let authors
+        typeof pkg.author === 'string' ? authors = [parseAuthor(pkg.author).name] : authors = [pkg.author.name]
```
Comment (malept) [inline src/installer.js]: Did this not work?
```suggestion
        const authors = [typeof pkg.author === 'string' ? parseAuthor(pkg.author).name : pkg.author.name]
```

File: src/installer.js [hunk 2]
```diff
@@ -118,7 +118,8 @@ class SquirrelInstaller extends common.ElectronInstaller {
     return common.readMetadata(this.userSupplied)
       .then(pkg => {
         pkg = pkg || {}
-        const authors = [parseAuthor(pkg.author).name]
+        let authors
```
Comment (malept) [inline src/installer.js]:
```suggestion
```

**Comment by malept** [comment]
Still needs a regression test?

**Comment by fcastilloec** [comment]
I created another PR for the test, check #258 Hopefully that would be enough of a regression test.

**Pull request merged**